pkgpanda/docs/create_and_maintain_packages.md

```diff
@@ -0,0 +1,79 @@
+* `url_extract`: URL to an artifact that is a compressed file which will be automatically decompressed before volume mounting. Supported compression types are: `.tar.gz`, `.tgz`, and `.zip`. Required fields are: `kind`, `url`, and `sha1`
+
+Rules for declaring sources in `buildinfo.json`:
+* If there are no required dependencies outside of the package directory, then no source information is required in `buildinfo.json`
+* If there is only one source, a `single_source` field must be included in the `buildinfo.json`. The data from this source will be mounted inside the container as `/pkg/src/$PKG_NAME`
+* If there are multiple sources, a `sources` field must be provided in `buildinfo.json` and each of its subfields will be a source like the single source entry. Each of these sources will be mounted in the container as `/pkg/src/<<key name of source in JSON sources>>`
+
+### Customized Docker Build Images
+
+By default, all packages are built with the [dcos-builder Dockerfile](../pkgpanda/docker/dcos-builder). A different docker image can be specified. There are two ways to specify a different docker image:
+* include a directory called `docker` alongside the `build` and `buildinfo.json`. This file must contain a Dockerfile and any necessary data to be included with the image. This docker image will be built as part of the package building process and will be named as `dcos/dcos-builder:<<package_name>>_dockerdir-latest` where `<<package_name>>` is the package being built.
+* reference a public DockerHub image in the `buildinfo.json`
+
+### Magic Paths and Variables in Build Containers
+* `/pkg/src/`: directory for all sources declared in `buildinfo.json`
+* `/pkg/extra/`: mounted `extra` directory from the package folder
+* `/opt/mesosphere/environment.export`: file including the environment variables that will link the build container environment to built packages and their artifacts. Most builds depending on other packages should source the environment with this file
+* `/opt/mesosphere`: the partially constructed DC/OS installation directory as it will be unpacked on hosts
+* `/opt/mesosphere/packages`: the location of all packages that were listed as dependencies.
+* `/opt/mesosphere/lib`: combined lib path of exported artifacts from the depedent packages. I.E. the data transferred to `$PKG_PATH/lib` combined across packages
+* `/opt/mesosphere/lib`: combined bin path of exported artifacts from the depedent packages. I.E. the data transferred to `$PKG_PATH/bin` combined across packages
+* `$PKG_NAME`: the name of the package folder
+* `$PKG_PATH`: the writable path for build artifacts
+
+# Guiding principles when packaging
+
+## Depending on other packages
+
+Every package lives in isolation from other packages, and can be installed/uninstalled atomically.
+
+Packages are unpacked to a metadata file `pkginfo.json`, and Every package has a *Package Name* as well as a *Package
```

Comment:
Neither am I. Its not super apparent because I was lazy in my git history, but this is from the old "for_packager.md" document as were a lot of the contentious issues here